src/content/blog/2022-08-03-new-config-system-part-1.md

```diff
@@ -0,0 +1,130 @@
+
+### The `overrides` key
+
+ESLint continued to receive requests for more powerful ways to configure their projects. More specifically, there were requests to provide [glob-based configs](https://github.com/eslint/eslint/issues/3611) from within existing config files. This led to the creation of an `overrides` key that would let you further modify configurations for a specific subset of files that ESLint was linting. Here's an example:
+
+```json
+{
+  "rules": {
+    "quotes": ["error", "double"]
+  },
+
+  "overrides": [
+    {
+      "files": ["bin/*.js", "lib/*.js"],
+      "excludedFiles": "*.test.js",
+      "rules": {
+        "quotes": ["error", "single"]
+      }
+    }
+  ]
+}
+```
+
+In this case, the JavaScript files in `bin` and `lib` prefer single quotes instead of the double quotes preferred everywhere else.
+
+The `overrides` key with glob-based configuration turned out to be a much better way of accomplishing what the configuration cascade was attempting to do. Once again, in hindsight, this would have been the perfect time to try and remove the cascade...but we didn't. And the complexity didn't stop there.
+
+### Adding `extends` to `overrides`
+
+The last step of eslintrc development was to add the `extends` key to `overrides` configurations, allowing users to inject additional config data into a glob-based config object, like this:
+
+```json
+{
+  "rules": {
+    "quotes": ["error", "double"]
+  },
+
+  "overrides": [
+    {
+      "files": ["bin/*.js", "lib/*.js"],
+      "excludedFiles": "*.test.js",
+      "extends": ["eslint:recommended"],
+      "rules": {
+        "quotes": ["error", "single"]
+      }
+    }
+  ]
+}
+```
+
+This addition also introduced a lot of additional complexity because we had to figure out how to merge glob patterns between two different configs. The end result was that `extends` inside of an `overrides` config would use an AND operator to merge `files` and `excludedFiles`. If you're not sure what exactly that means, you're not alone. It's confusing even to us.
+
+## The need for simplification
+
+Around the new year of 2019, I was getting more concerned about the complexity of the eslintrc system. We were getting more and more questions about obscure error messages related to loading config files that couldn't find other config files or plugins. Additionally, the team was collectively becoming afraid of touching anything to do with the config system. No one really understood all of the different permutations around calculating the final config for any given file. We had fallen into the trap that may software projects do: we kept adding new features without taking a a step back to look at the problem (and solution) holistically. This had led to an almost unmaintainable part of our codebase.
```

Comment:
Again, I would avoid "I" here, since it would not be clear to end use who is writing this blog post.

Reply:
I would also mentioned that there was a continues stream of requests for shareable configs to be able to include plugins as a bundle, and without peerDependencies we didn't have a solution for this. Also might be worth including the fact that the team felt that you should be able to configure multiple versions of the same rule from a plugin, which could not be easily solved by using `require` approach that config was using for `extends`.